### PR TITLE
SONARSLANG 47 Setup integration tests for plugin

### PR DESCRIPTION
**Pull request by christophe-zurn-sonarsource**

### CODE REVIEW

File: its/plugin/src/test/java/com/sonarsource/slang/DefaultProfileTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.Test;
+import org.sonarqube.ws.QualityProfiles.SearchWsResponse.QualityProfile;
+import org.sonarqube.ws.Rules;
+import org.sonarqube.ws.client.qualityprofile.SearchWsRequest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DefaultProfileTest extends TestBase {
```
Comment (pynicolas) [inline its/plugin/src/test/java/com/sonarsource/slang/DefaultProfileTest.java]: I'm not sure that this test class is really useful.

File: its/plugin/src/test/java/com/sonarsource/slang/FileSuffixesTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang;
+
+import com.sonar.orchestrator.build.SonarScanner;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FileSuffixesTest extends TestBase {
```
Comment (pynicolas) [inline its/plugin/src/test/java/com/sonarsource/slang/FileSuffixesTest.java]: I'm not sure that this test class is really useful.

File: its/plugin/src/test/java/com/sonarsource/slang/MeasuresTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  private static final String BASE_DIRECTORY = "projects/measures/";
+
+  @Test
+  public void kotlin_measures() {
+    ORCHESTRATOR.executeBuild(getSonarScanner(BASE_DIRECTORY, "kotlin"));
+
+    assertThat(getMeasureAsInt("empty_file.kt", "lines")).isEqualTo(1);
+    assertThat(getMeasureAsInt("file1.kt", "lines")).isEqualTo(14);
+    assertThat(getMeasureAsInt("file2.kt", "lines")).isEqualTo(12);
```
Comment (pynicolas) [inline its/plugin/src/test/java/com/sonarsource/slang/MeasuresTest.java]: "lines" metric is the responsibility of SQ, and I don't think we should test it.

File: slang-api/src/main/java/com/sonarsource/slang/visitors/TreeVisitor.java [hunk 4]
```diff
@@ -34,14 +34,12 @@ public TreeVisitor() {
...
-    if (root != null) {
-      ctx.before(root);
-      before(ctx, root);
-      visit(ctx, root);
-      after(ctx, root);
-    }
+    ctx.before(root);
+    before(ctx, root);
+    visit(ctx, root);
+    after(ctx, root);
```
Comment (pynicolas) [inline slang-api/src/main/java/com/sonarsource/slang/visitors/TreeVisitor.java]: How is this change related to integration tests?

Comment (christophe-zurn-sonarsource) [reply slang-api/src/main/java/com/sonarsource/slang/visitors/TreeVisitor.java]: This was related to the empty_file.kt which resulted in a parsing error and had no metrics.
Reverted these changes in 1f44432452b84a70f79748bd21019f97ee7e0e85, as discussed, we consider it normal not to have measures for files with parsing errors

File: its/plugin/src/test/java/com/sonarsource/slang/MeasuresTest.java (hunk 3 above)
Comment (christophe-zurn-sonarsource) [reply its/plugin/src/test/java/com/sonarsource/slang/MeasuresTest.java]: Ok. Removed it in 1f44432452b84a70f79748bd21019f97ee7e0e85

File: its/plugin/src/test/java/com/sonarsource/slang/FileSuffixesTest.java (hunk 2 above)
Comment (christophe-zurn-sonarsource) [reply its/plugin/src/test/java/com/sonarsource/slang/FileSuffixesTest.java]: I would keep it for now, as it still tests the default behaviour of plugin when a empty file suffix is provided

File: its/plugin/src/test/java/com/sonarsource/slang/DefaultProfileTest.java (hunk 1 above)
Comment (christophe-zurn-sonarsource) [reply its/plugin/src/test/java/com/sonarsource/slang/DefaultProfileTest.java]: Ok. Removed it in 1f44432452b84a70f79748bd21019f97ee7e0e85

File: Jenkinsfile [hunk 5]
```diff
@@ -69,13 +78,25 @@ pipeline {
...
+    }
+  }
+}
+
+def runPlugin(String sqRuntimeVersion) {
   withQAEnv {
     withMaven(maven: MAVEN_TOOL) {
       mavenSetBuildVersion()
       dir('its') {
         sh 'git submodule update --init --recursive'
```
[the remark this answers is not in the document]
Comment (christophe-zurn-sonarsource) [reply Jenkinsfile]: Indeed, we don't need it for the plugin ITs

File: Jenkinsfile [hunk 6]
```diff
@@ -23,30 +23,39 @@ pipeline {
     }
     stage('QA') {
       parallel {
-        stage('ITs-lts') {
+        stage('ruling-lts') {
```
[the remark this answers is not in the document]
Comment (christophe-zurn-sonarsource) [reply Jenkinsfile]: Good point. Done in 1f44432452b84a70f79748bd21019f97ee7e0e85
